src/Network.cpp

```diff
@@ -0,0 +1,213 @@
+                mCallback->onRequestGameOperation();
+                break;
+            case spy::network::messages::MessageTypeEnum::STATISTICS:
+                //TODO implement with validation check (playerId, sessionId, ..)
+                mCallback->onStatistics();
+                break;
+            case spy::network::messages::MessageTypeEnum::GAME_LEFT:
+                //TODO implement with validation check (playerId, sessionId, ..)
+                mCallback->onGameLeft();
+                break;
+            case spy::network::messages::MessageTypeEnum::GAME_PAUSE:
+                //TODO implement with validation check (playerId, sessionId, ..)
+                mCallback->onGamePause();
+                break;
+            case spy::network::messages::MessageTypeEnum::META_INFORMATION:
+                //TODO implement with validation check (playerId, sessionId, ..)
+                mCallback->onMetaInformation();
+                break;
+            case spy::network::messages::MessageTypeEnum::STRIKE:
+                //TODO implement with validation check (playerId, sessionId, ..)
+                mCallback->onStrike();
+                break;
+            case spy::network::messages::MessageTypeEnum::ERROR:
+                //TODO implement with validation check (playerId, sessionId, ..)
+                mCallback->onError();
+                break;
+            case spy::network::messages::MessageTypeEnum::REPLAY:
+                //TODO implement with validation check (playerId, sessionId, ..)
+                mCallback->onReplay();
+                break;
+            default:
+                // do nothing
+                break;
+        }
+    }
+
+    void Network::connect(const std::string &servername, int port) {
+        websocket::network::WebSocketClient client(servername, "/", port, "");
+
+        this->mWebSocketClient.emplace(servername, "/", port, "");
+        mWebSocketClient->receiveListener.subscribe(std::bind(&Network::onReceiveMessage, this, std::placeholders::_1));
+        mWebSocketClient->closeListener.subscribe(std::bind(&Network::disconnect, this));
+    }
+
+    void Network::disconnect() {
+        // TODO validation check (rules of game, ...)
+        if (mWebSocketClient.has_value()) {
+            mWebSocketClient.reset();
+        }
+        //TODO implement (reset model, ...)
+    }
+
+    bool Network::sentHello(std::string name, spy::network::RoleEnum role) {
+        if (role == spy::network::RoleEnum::INVALID) {
+            return false;
+        }
+        //TODO validation check (rules of game, ...)
+        auto message = spy::network::messages::Hello(mModel->clientState.id, std::move(name), role);
+        nlohmann::json j = message;
+        mWebSocketClient->send(j.dump());
+        return true;
+    }
+
+    bool Network::sentItemChoice(std::variant<spy::util::UUID, spy::gadget::GadgetEnum> choice) {
+        if (mModel->clientState.role == spy::network::RoleEnum::INVALID ||
```

Comment:
Kann man irgendwie sicherstellen dass dieses Objekt nie INVALID ist? das ist ja arg umständlich das sonst immer zu testen...

Reply:
Generell sollte der Test nicht hier rein, sondern in LibCommon ([siehe Issue #12](https://github.com/SoPra-Team-17/LibCommon/issues/12)). Aber ja wäre natürlich cool, wenn das gehen würde.

Reply:
Vllt kann man da irgendwas mit Templates machen...